src/Client/ClientFactory.php

```diff
@@ -27,6 +33,19 @@ public function createFromApiCredentials(): AkeneoPimClientInterface
 
         $client = new AkeneoPimClientBuilder($apiConnection->getBaseUrl());
 
+        $httpClient = HttpClient::create();
+
+        $path = $this->cacheDir . '/akeneo';
+
+        if (is_dir($path) === false) {
+            mkdir($path);
+        }
+
+        $store = new Store($path);
+        $httpClient = new CachingHttpClient($httpClient, $store, ['default_ttl' => 3600, 'allow_revalidate' => true, 'debug' => getenv('APP_DEBUG')]);
+
+        $client->setHttpClient(new HttplugClient($httpClient));
```

Comment:
Oh, maybe akeneo, change the contact since my pr is open ...

Reply:
I confirm... Akeneo change that : 

https://github.com/akeneo/api-php-client/commit/e41246a0ed8f7a6f5f9b0d1acf38a664d740f639

Reply:
I think we can use the integrated akeneo cache system that creates a CachedResourceClient object to store resources.

$client = new AkeneoPimClientBuilder($apiConnection->getBaseUrl());
$client->enableCache();

Reply:
I think is not a good idea, because he use memory cache, and if you have some data, you risk to throw memory limit.

With my system, cache use is file system.

Reply:
thanks @baiiko. We will check if we have to block the 11.3.0 version to benefits from the possibility to implement caching as you did. I think we should provide a way to store it to different places (like redis...) to be able to use the same cache if we are working on multiple server nodes to process the queue.

Reply:
You can talk about @oallain if necessary, he can talk with me directly :)